common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -541,16 +491,22 @@ func (v *pinotVisibilityTripleManager) CountWorkflowExecutions(
 		closeStatus: 6, // 6 means not set closeStatus.
 		customQuery: request.Query,
 	}, request.Domain)
-	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	manager := v.chooseVisibilityManagerForRead(ctx, request.Domain)
+	return manager.CountWorkflowExecutions(ctx, request)
+}
+
+func (v *pinotVisibilityTripleManager) chooseVisibilityManagerForRead(ctx context.Context, domain string) VisibilityManager {
+	// temporary code: for Pinot migration usage -->
 	if override := ctx.Value(ContextKey); override == Primary {
-		manager = v.esVisibilityManager
+		v.logger.Debug("Pinot Migration log: Primary visibility manager was chosen for read.")
+		return v.esVisibilityManager
 	} else if override == Secondary {
-		manager = v.pinotVisibilityManager
+		v.logger.Debug("Pinot Migration log: Secondary visibility manager was chosen for read.")
+		return v.pinotVisibilityManager
 	}
-	return manager.CountWorkflowExecutions(ctx, request)
-}
+	v.logger.Debug("Pinot Migration log: ContextKey was empty.")
+	// temporary code: ends here <--
```

Comment:
probably better to just link to a task ID or a todo

Reply:
Created a Jira ticket to record this.